mp2-v1/tests/common/cases/table_source.rs

```diff
@@ -1436,3 +1460,416 @@ fn evm_word_column_info(table_info: &[ColumnInfo]) -> Vec<SlotEvmWordColumns> {
+                                ColumnMetadata::PrimaryKey(_) => true,
+                                ColumnMetadata::NoPrimaryKey(_) => false,
+                            });
+                        let columns_to_update = if primary_key_columns.is_empty() {
+                            // no primary key columns among non-indexed columns,
+                            // so we update 2 columns at random
+                            [0; 2].map(|_| {
+                                non_primary_key_columns
+                                    [rng.gen_range(0..non_primary_key_columns.len())]
+                                .clone()
+                            })
+                        } else {
+                            [
+                                non_primary_key_columns
+                                    [rng.gen_range(0..non_primary_key_columns.len())]
+                                .clone(),
+                                primary_key_columns[rng.gen_range(0..primary_key_columns.len())]
+                                    .clone(),
+                            ]
+                        };
+                        rows_to_update
+                            .into_iter()
+                            .zip(columns_to_update)
+                            .flat_map(|(row_index, column_data)| {
+                                self.update_non_indexed_column(row_index, &column_data)
+                                    .unwrap()
+                            })
+                            .collect()
+                    }
+                }
+            }
+            ChangeType::Silent => vec![], // no changes
+        }
+    }
+
+    pub(crate) fn generate_extraction_proof_inputs(
+        &self,
+        ctx: &mut TestContext,
+        proof_key: ProofKey,
+    ) -> Result<(ExtractionProofInput, HashOutput)> {
+        // convert current row values to row cells
+        let rows = self
+            .row_values
+            .iter()
+            .map(CellCollection::try_from)
+            .collect::<Result<Vec<_>>>()?;
+        let rng = &mut thread_rng();
+        // This could be computed from the table data according to any logic,
+        // here for simplicity we just generate it at random
+        let hash = HashOutput::from(array::from_fn(|_| rng.gen()));
+        // fetch previous hash from IVC proof for previous primary index, if any
+        let prev_hash = {
+            if let Ok(proof) = ctx.storage.get_proof_exact(&proof_key) {
+                // fetch previous hash from the proof
+                let pproof = ProofWithVK::deserialize(&proof)?;
+                let pis = PublicInputs::from_slice(&pproof.proof().public_inputs);
+                pis.block_hash_output()
+            } else {
+                // there is no previous primary index, so we generate previous hash at random
+                HashOutput::from(array::from_fn(|_| rng.gen()))
```

Comment:
Even if there is, could we avoid relying on this ?

Reply:
AFAIU this is asking the same as [this comment](https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/415#discussion_r1946123252)?

Reply:
This has been removed after the API changes in commit https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/415/commits/2f71b25f22f08043f029b17af38adf20bbe76faf